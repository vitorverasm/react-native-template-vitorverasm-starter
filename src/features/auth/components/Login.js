import React, { Component } from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import * as routes from '../../../navigation/routes';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  onPress = () => {
    const { navigation } = this.props;
    navigation.navigate(routes.HOME);
  };

  render() {
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <TouchableOpacity onPress={this.onPress}>
          <Text>Login screen</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default Login;
