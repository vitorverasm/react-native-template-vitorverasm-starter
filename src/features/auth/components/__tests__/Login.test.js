import { shallow } from 'enzyme';
import React from 'react';
import Login from '../Login';

describe('Login.js', () => {
  const props = {
    navigation: {
      navigate: jest.fn
    }
  };
  it('renders correctly', () => {
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('onPress[function]', () => {
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper.instance().onPress()).not.toBeNull();
  });
});
