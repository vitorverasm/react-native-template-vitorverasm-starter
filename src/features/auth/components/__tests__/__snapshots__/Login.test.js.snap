// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.js renders correctly 1`] = `ShallowWrapper {}`;
